### PR TITLE
20.0.0.9 release blog post

### PR DESCRIPTION
**Pull request by lauracowen**
Please review: https://draft-openlibertyio.mybluemix.net/blog/2020/08/28/graphql-apis-open-liberty-20.0.0.9.html

@andymc12 Could you double-check the wording of the title and first paragraph of the post as well as your sections, please?

2020-08-27
image

2020-08-26
image

### CODE REVIEW

File: posts/2020-08-28-graphql-apis-with-open-liberty-20.0.0.9.adoc [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    <classloader apiTypeVisibility="+third-party"/>
+</application>
+----
+Now you can access SmallRye GraphQL APIs like the type-safe client!  Note that these APIs may change in future releases as SmallRye is constantly evolving.  For more information on the SmallRye GraphQL project, please visit https://github.com/smallrye/smallrye-graphql
+The SmallRye GraphQL Client APIs are very similar to https://github.com/eclipse/microprofile-rest-client[MicroProfile Rest Client] in that it uses an interface to invoke remote methods in a type-safe manner. For example, suppose we want a client that can invoke a query of all super heroes in a given location.  We would create a query interface like this:
+[source,java]
+----
+@GraphQlClientApi
+interface SuperHeroesApi {
+    List<SuperHero> allHeroesIn(String location); // (1)
```
Comment (andymc12) [inline posts/2020-08-28-graphql-apis-with-open-liberty-20.0.0.9.adoc]:
```suggestion
    List<SuperHero> allHeroesIn(String location);
```

removing the unnecessary comment - sorry, I had left that in there from the example code in my original post.

File: posts/2020-08-28-graphql-apis-with-open-liberty-20.0.0.9.adoc [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+== Experiment with a third-party GraphQL client API
+
+MicroProfile GraphQL has only been available in Open Liberty for a few months now, and it is already a hit! That said, there are lots of things we'd like to do to improve it - and make it more complete. One of those things is a Client API. While the official Client API is not expected until the next release of MicroProfile GraphQL, you can now experiment with the type-safe SmallRye GraphQL Client API. SmallRye is the underlying implementation of MicroProfile GraphQL, and you can access some of its above-and-beyond-the-spec features by adding the "third-party" API type visibility to your application, like so:
+[source,xml]
+----
+<application name="MyGraphQLApp" location="MyGraphQLApp.war">
+    <classloader apiTypeVisibility="+third-party"/>
+</application>
+----
+Now you can access SmallRye GraphQL APIs like the type-safe client!  Note that these APIs may change in future releases as SmallRye is constantly evolving.  For more information on the SmallRye GraphQL project, please visit https://github.com/smallrye/smallrye-graphql
```
Comment (andymc12) [inline posts/2020-08-28-graphql-apis-with-open-liberty-20.0.0.9.adoc]:
```suggestion
Now you can access SmallRye GraphQL APIs like the type-safe client!  Note that these APIs may change in future releases as SmallRye is constantly evolving.  For more information on the SmallRye GraphQL project, please visit https://github.com/smallrye/smallrye-graphql - in Open Liberty 20.0.0.9, we are using SmallRye GraphQL 1.0.7.

```

Updating to include the version of SmallRye that we ship - and adding a line break to (hopefully) separate the two paragraphs.

File: posts/2020-08-28-graphql-apis-with-open-liberty-20.0.0.9.adoc [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+----
+Where `SuperHero` on the client side looks like this:
+[source,java]
+----
+class SuperHero {
+    private String name;
+    private List<String> superPowers;
+}
+----
+The `SuperHero` entity might contain dozens more fields, but if we're only interested in the hero's name and super powers, then we only need those two fields in our client-side class.  Now we can invoke the query with some code like this:
```
Comment (andymc12) [inline posts/2020-08-28-graphql-apis-with-open-liberty-20.0.0.9.adoc]:
```suggestion
The `SuperHero` entity might contain dozens more fields on the server side, but if we're only interested in the hero's name and super powers, then we only need those two fields in our client-side class.  Now we can invoke the query with some code like this:
```

File: posts/2020-08-28-graphql-apis-with-open-liberty-20.0.0.9.adoc [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+//end::run[]
+Or take a look at our link:{url-prefix}/downloads/[Downloads page].
+[link=https://stackoverflow.com/tags/open-liberty]
+image::img/blog/blog_btn_stack.svg[Ask a question on Stack Overflow, align="center"]
+
+//tag::features[]
+[#GraphQLAPIs]
+== Experiment with a third-party GraphQL client API
+
+MicroProfile GraphQL has only been available in Open Liberty for a few months now, and it is already a hit! That said, there are lots of things we'd like to do to improve it - and make it more complete. One of those things is a Client API. While the official Client API is not expected until the next release of MicroProfile GraphQL, you can now experiment with the type-safe SmallRye GraphQL Client API. SmallRye is the underlying implementation of MicroProfile GraphQL, and you can access some of its above-and-beyond-the-spec features by adding the "third-party" API type visibility to your application, like so:
```
Comment (andymc12) [inline posts/2020-08-28-graphql-apis-with-open-liberty-20.0.0.9.adoc]:
```suggestion
https://github.com/eclipse/microprofile-graphql[MicroProfile GraphQL] has only been available in Open Liberty for a few months now, and it is already a hit! That said, there are lots of things we'd like to do to improve it - and make it more complete. One of those things is a Client API. While the official Client API is not expected until the next release of MicroProfile GraphQL, you can now experiment with the type-safe SmallRye GraphQL Client API. SmallRye is the underlying implementation of MicroProfile GraphQL, and you can access some of its above-and-beyond-the-spec features by adding the "third-party" API type visibility to your application, like so:
```

File: posts/2020-08-28-graphql-apis-with-open-liberty-20.0.0.9.adoc [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+The `SuperHero` entity might contain dozens more fields, but if we're only interested in the hero's name and super powers, then we only need those two fields in our client-side class.  Now we can invoke the query with some code like this:
+[source,java]
+----
+SuperHeroesApi api = GraphQlClientBuilder.newBuilder().build(SuperHeroesApi.class);
+List<SuperHero> heroesOfNewYork = api.allHeroesIn("NYC");
+----
+Remember that this client API is not official, but the official API (in MicroProfile GraphQL 1.1) will be based on this - think of this as a sneak peak!
+
+[#GraphiQL]
+== Built-in GraphiQL user interface for faster queries and mutations
```
Comment (andymc12) [inline posts/2020-08-28-graphql-apis-with-open-liberty-20.0.0.9.adoc]:
```suggestion
== Built-in GraphiQL user interface for faster development of queries and mutations
```

The UI is intended to make it easier and faster for developers to create and test queries and mutation.

File: posts/2020-08-28-graphql-apis-with-open-liberty-20.0.0.9.adoc [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+Develop MicroProfile GraphQL queries and mutations faster with the built-in GraphiQL user interface!
+Open Liberty now sports a built-in https://github.com/graphql/graphiql/blob/main/packages/graphiql/README.md[GraphiQL] user interface. This web-based UI allows you to write and execute GraphQL queries and mutations in real time with advanced editing features like command completion, query history, schema introspection, etc.
+To enable the UI, you must first https://openliberty.io/blog/2020/06/10/microprofile-graphql-open-liberty.html[write and deploy a MicroProfile GraphQL application].  Then add this line to your server.xml:
```
Comment (andymc12) [inline posts/2020-08-28-graphql-apis-with-open-liberty-20.0.0.9.adoc]:
```suggestion

To enable the UI, you must first https://openliberty.io/blog/2020/06/10/microprofile-graphql-open-liberty.html[write and deploy a MicroProfile GraphQL application].  Then add this line to your server.xml:
```

Suggested line break to separate this paragraph.

File: posts/2020-08-28-graphql-apis-with-open-liberty-20.0.0.9.adoc [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+
+* Resolved in issue link:https://github.com/OpenLiberty/open-liberty/issues/13138[#13138]
+
+
+=== MicroProfile Graphql Exception allowlist now up and running 
+
+As an open source product, we love hearing feedback from Open Liberty users.  We especially love comments like this: "Hello, I am using microprofile-graphql on openliberty and everything goes well…" and even the rest of that sentence (taken from Open Liberty https://github.com/OpenLiberty/open-liberty/issues/13036[#13036]): "except for the exception whitelisting mechanism via microprofile config described here."
+Our MicroProfile GraphQL feature has only been generally available for a few months, so it's great to see that users are using it - and it's exciting to see that they are already exploring the "dark corners" (exception handling, etc.).  While we hate to find out that we let a bug slip through the cracks, we're always eager to fix them when they do.
```
Comment (andymc12) [inline posts/2020-08-28-graphql-apis-with-open-liberty-20.0.0.9.adoc]:
```suggestion

Our MicroProfile GraphQL feature has only been generally available for a few months, so it's great to see that users are using it - and it's exciting to see that they are already exploring the "dark corners" (exception handling, etc.).  While we hate to find out that we let a bug slip through the cracks, we're always eager to fix them when they do. 

```

Suggested line breaks to separate paragraphs.

**Review by andymc12 — COMMENTED** [review]
Almost all comments are issues with the text I originally sent you - so, sorry about that.  I added some spacing to hopefully separate paragraphs - and one issue with a section header.  Otherwise, looks good.  Thanks!

**Comment by lauracowen** [comment]
Some minor comments while I remember before I review it properly:
- remove "with" from the file name
- make the "Resolved in…" sentence the last sentence of the paragraph, not as a separate paragraph/bullet.

File: posts/2020-08-28-graphql-apis-with-open-liberty-20.0.0.9.adoc (hunk 1 above)
Comment (jakub-pomykala) [reply posts/2020-08-28-graphql-apis-with-open-liberty-20.0.0.9.adoc]: Done

File: posts/2020-08-28-graphql-apis-with-open-liberty-20.0.0.9.adoc (hunk 2 above)
Comment (jakub-pomykala) [reply posts/2020-08-28-graphql-apis-with-open-liberty-20.0.0.9.adoc]: Changed to "Now you can access SmallRye GraphQL APIs like the type-safe client! Note that these APIs may change in future releases as SmallRye is constantly evolving. For more information please visit https://github.com/smallrye/smallrye-graphql[SmallRye GraphQL project] - in Open Liberty 20.0.0.9, we are using SmallRye GraphQL 1.0.7." instead

File: posts/2020-08-28-graphql-apis-with-open-liberty-20.0.0.9.adoc (hunk 3 above)
Comment (jakub-pomykala) [reply posts/2020-08-28-graphql-apis-with-open-liberty-20.0.0.9.adoc]: done

File: posts/2020-08-28-graphql-apis-with-open-liberty-20.0.0.9.adoc (hunk 4 above)
Comment (jakub-pomykala) [reply posts/2020-08-28-graphql-apis-with-open-liberty-20.0.0.9.adoc]: done

File: posts/2020-08-28-graphql-apis-with-open-liberty-20.0.0.9.adoc (hunk 6 above)
Comment (jakub-pomykala) [reply posts/2020-08-28-graphql-apis-with-open-liberty-20.0.0.9.adoc]: Done

File: posts/2020-08-28-graphql-apis-with-open-liberty-20.0.0.9.adoc (hunk 5 above)
Comment (jakub-pomykala) [reply posts/2020-08-28-graphql-apis-with-open-liberty-20.0.0.9.adoc]: Done

File: posts/2020-08-28-graphql-apis-with-open-liberty-20.0.0.9.adoc (hunk 7 above)
Comment (jakub-pomykala) [reply posts/2020-08-28-graphql-apis-with-open-liberty-20.0.0.9.adoc]: Done

**Review by andymc12 — APPROVED** [review]
LGTM - thanks!

**Comment by jakub-pomykala** [comment]
Updated authors info

**Comment by lauracowen** [comment]
Some last minor style things:
- Can you update the first two sentences of the descriptions in the front matter (metadata) to be along the same lines as the intro paragraph of the blog post? I missed that when I edited the intro.
- For the GraphiQL section, I know I gave you a different heading but having the first line as a separate paragraph like that looks odd and repetitive. Just remove the heading I gave you and make that first line into the heading (it wasn't formatted correctly before so it seemed weird as the heading but now I think it'd work ok). There's no point having both.
- Can you make the GraphiQL URL into monospace instead of a clickable link (just put backticks around it) as it's not a valid URL on the website.
- Can you change "-  for example, suppose we use" to start a new sentence instead: "For example, suppose we use". And put a full-stop after the closing backtick of the previous word (so the full-stop isn't highlighted in monospace).
- "JAX-RS clients will now be able to" -> "JAX-RS clients can now" (easier to read)
- a typo I think? "has been added to that a single internal provider" - maybe should be "so that"?
- In the featureUtility description, remove "This bug was caused where " and start with "The `include` element was being ignored by the featureUtility comand.".
- Change "The include element can be used in a server.xml in order to consolidate a configuration from a separate server.xml file. " to "The `include` element can be used in a `server.xml` file to consolidate configuration from additional separate `server.xml` files. "
- Make "opening an issue on GitHub" into a link to the URL that follows (and remove "at URL" after that sentence. 
- Change "reach us via Twitter" to "on Twitter"
- Add a full-stop after the second Gitter link.
- if you haven't already, quickly go through clicking each link to make sure it works and goes to the right place (or at least makes sense based on the text).

Thanks

File: posts/2020-08-28-graphql-apis-open-liberty-20.0.0.9.adoc [hunk 8]
```diff
@@ -0,0 +1,140 @@
+---
+layout: post
+title: "Quicker, easier GraphQL queries with Open Liberty 20.0.0.9"
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/jakub-pomykala
+author_github: https://github.com/jakub-pomykala
+seo-title: Using MicroProfile GraphQL third-party APIs, using the built-in user interface for GraphiQL, plus significant bug fixes on Open Liberty 20.0.0.9 - OpenLiberty.io
+seo-description: With Open Liberty 20.0.0.9 you can now Experiment with the type-safe SmallRye GraphQL Client API. Try writting and running GraphQL queries and mutations using a built-in GraphiQL user interface. Plus a number of significant bug fixes.
+blog_description: With Open Liberty 20.0.0.9 you can now Experiment with the type-safe SmallRye GraphQL Client API. Try writting and running GraphQL queries and mutations using a built-in GraphiQL user interface. Plus a number of significant bug fixes.
```
Comment (mbroz2) [inline posts/2020-08-28-graphql-apis-open-liberty-20.0.0.9.adoc:9]: add `open-graph-image: https://openliberty.io/img/blog/GraphiQL_UI.png`

File: posts/2020-08-28-graphql-apis-open-liberty-20.0.0.9.adoc [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+[source,xml]
+----
+<application name="MyGraphQLApp" location="MyGraphQLApp.war">
+    <classloader apiTypeVisibility="+third-party"/>
+</application>
+----
+Now you can access SmallRye GraphQL APIs like the type-safe client! Note that these APIs may change in future releases as SmallRye is constantly evolving. For more information please visit https://github.com/smallrye/smallrye-graphql[SmallRye GraphQL project] - in Open Liberty 20.0.0.9, we are using SmallRye GraphQL 1.0.7.
+The SmallRye GraphQL Client APIs are very similar to https://github.com/eclipse/microprofile-rest-client[MicroProfile Rest Client] in that it uses an interface to invoke remote methods in a type-safe manner. For example, suppose we want a client that can invoke a query of all super heroes in a given location. We would create a query interface like this:
```
Comment (mbroz2) [inline posts/2020-08-28-graphql-apis-open-liberty-20.0.0.9.adoc]: I'd rather [MicroProfile Rest Client] point at our MP Rest Client guide (https://openliberty.io/guides/microprofile-rest-client.html)

**Review by mbroz2 — APPROVED** [review]

**Review by mbroz2 — APPROVED** [review]

**Pull request merged**